Validate datasets are found when checking bigquery

snoe opened the pull request:
Fixes #19709

BigQuery `can-connect?` will now throw with a message `Looks like we cannot find any matching datasets.` This is only checked when `Only these` or `All except` are selected (not checked when `All` datasets are chosen).

Made a general mechanism to pass expected messages to users in
api/database via ex-info. This allows us to suppress logging for
"unexceptional" exceptions that one can expect to hit while setting up
drivers.

snoe committed:
Validate datasets are found when checking bigquery

Fixes #19709

dpsutton commented:
<img width="1310" alt="image" src="https://user-images.githubusercontent.com/6377293/165590127-ae2ed2e9-95a9-47e2-bf08-7d64c189dbfe.png">

snoe committed:
Address PR feedback

Made a general mechanism to pass expected messages to users in
api/database via ex-info. This allows us to suppress logging for
"unexceptional" exceptions that one can expect to hit while setting up
drivers.

snoe committed:
Only validate when filters are set

Also removed the dataset list from the exception as it's not surfaced to
users.
